onnxruntime/core/providers/xnnpack/math/gemm.cc

```diff
@@ -0,0 +1,193 @@
+// Copyright(c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT License.
+
+#include "gemm.h"
+#include "core/framework/transpose_helper.h"
+#include "core/providers/utils.h"
+
+namespace onnxruntime {
+namespace xnnpack {
+
+bool Gemm::IsGemmOnnxNodeSupported(const NodeUnit& node_unit, const GraphViewer& graph) {
+  bool supported = false;
+  const onnxruntime::Node& node = node_unit.GetNode();
+
+  // use do {} while(false) so it's easier to set a breakpoint on the return
+  do {
+    ConstPointerContainer<std::vector<NodeArg*>> input_defs = node.InputDefs();
+
+    const auto alpha = node.GetAttributes().find("alpha");
+    if (!(*alpha).second.has_f() && (*alpha).second.f() != 1.0) break;
```

Comment:
nit: onnx spec says `alpha` must be float so if the attribute is there, has_f will always be true and you don't need to check it. same for `beta`